packages/node-experimental/src/integrations/http.ts

```diff
@@ -0,0 +1,194 @@
+    // No need to instrument if we don't want to track anything
+    if (!this._breadcrumbs && this._tracing === false) {
+      return;
+    }
+
+    const client = getCurrentHub().getClient<NodePreviewClient>();
+    const clientOptions = client?.getOptions();
+
+    this._shouldCreateSpans = typeof this._tracing === 'undefined' ? hasTracingEnabled(clientOptions) : this._tracing;
+
+    // Register instrumentations we care about
+    this._unload = registerInstrumentations({
+      instrumentations: [
+        new HttpInstrumentation({
+          requireParentforOutgoingSpans: true,
+          requireParentforIncomingSpans: false,
+          applyCustomAttributesOnSpan: (span, req, res) => {
+            this._onSpan(span as unknown as OtelSpan, req, res);
+          },
+        }),
+      ],
+    });
+
+    this._shouldCreateSpanForRequest =
+      // eslint-disable-next-line deprecation/deprecation
+      this._shouldCreateSpanForRequest || clientOptions?.shouldCreateSpanForRequest;
+
+    client?.otelHooks.on('spanEnd', this._onSpanEnd);
+  }
+
+  /**
+   *  Unregister this integration.
+   */
+  public unregister(): void {
+    this._unload?.();
+
+    const client = getCurrentHub().getClient<NodePreviewClient>();
+    client?.otelHooks.off('spanEnd', this._onSpanEnd);
+  }
+
+  private _onSpanEnd: (otelSpan: OtelSpan) => void | false = (otelSpan: OtelSpan) => {
+    if (!this._shouldCreateSpans) {
+      return false;
+    }
+
+    if (this._shouldCreateSpanForRequest) {
+      const url = getHttpUrl(otelSpan.attributes);
+      if (url && !this._shouldCreateSpanForRequest(url)) {
+        return false;
+      }
+    }
+
+    return undefined;
+  };
+
+  /** Handle an emitted span from the HTTP instrumentation. */
+  private _onSpan(
+    span: OtelSpan,
+    request: ClientRequest | IncomingMessage,
+    response: IncomingMessage | ServerResponse,
+  ): void {
+    const data = getRequestSpanData(span, request, response);
+    const { attributes } = span;
+
+    const additionalData: AdditionalOtelSpanData = {
+      tags: {},
+      data: {
+        url: data.url,
+      },
+      contexts: {},
+      metadata: {},
+    };
+
+    if (span.kind === SpanKind.SERVER) {
+      additionalData.metadata = { request };
+    }
+
+    if (attributes[SemanticAttributes.HTTP_STATUS_CODE]) {
+      const statusCode = attributes[SemanticAttributes.HTTP_STATUS_CODE] as string;
+      additionalData.tags['http.status_code'] = statusCode;
+      additionalData.data['http.response.status_code'] = statusCode;
+    }
+
+    if (data['http.query']) {
+      additionalData.data['http.query'] = data['http.query'].slice(1);
+    }
+    if (data['http.fragment']) {
+      additionalData.data['http.fragment'] = data['http.fragment'].slice(1);
+    }
+
+    addOtelSpanData(span.spanContext().spanId, additionalData);
+
+    getCurrentHub().addBreadcrumb(
```

Comment:
m: do we need to check if the `BreadCrumbs` integration is configured to record http breadcrumbs? 
(Not sure about this, I just know that I had to do this for client-side sveltekit fetch)

Reply:
Hmm not sure, if I look at the hub in core it has this method always 🤔 that delegates to scope.addBreadcrumb, which also seems to always be defined.

Reply:
Sorry, I mixed up the integrations, I rechecked and saw that the current `Http` integration has a `breadcrumbs` option which you added here as well.  

but shouldn't we nevertheless guard this line with `this._breadcrumbs`? afaict we only check `this._breadcrumbs` at the beginning to early return if both, breadcrumbs and tracing are disabled.